backend/tests/api/user_test.go

```diff
@@ -38,3 +41,93 @@ func TestGetAllUsersWorks(t *testing.T) {
 	// assert that the user returned from the database is the same as the user returned from the API
 	assert.Equal(dbUser, respUser)
 }
+
+func TestGetUserHappyPath(t *testing.T) {
+	// initialize the test
+	app, assert := InitTest(t)
+
+	// create a GET request to the APP/api/v1/users/:id endpoint
+	req := httptest.NewRequest("GET", fmt.Sprintf("%s/api/v1/users/1", app.Address), nil)
+
+	// send the request to the app
+	resp, err := app.App.Test(req)
+
+	assert.NilError(err)
+
+	assert.Equal(200, resp.StatusCode)
+
+	// decode the response body into a slice of users
+	var user *models.User
+
+	err = json.NewDecoder(resp.Body).Decode(&user)
+
+	assert.NilError(err)
+
+	dbUser, err := transactions.GetUser(app.Conn, "1")
+
+	assert.NilError(err)
+
+	assert.Equal(dbUser.Email, user.Email)
+	assert.Equal(user.Email, "generatesac@gmail.com")
+}
+
+func TestGetUserBadRequest(t *testing.T) {
+	app, assert := InitTest(t)
+	// create a GET request to the APP/api/v1/users/:id endpoint
+	req := httptest.NewRequest("GET", fmt.Sprintf("%s/api/v1/users/letters", app.Address), nil)
+	resp, err := app.App.Test(req)
+	assert.NilError(err)
+	defer resp.Body.Close()
+	bodyBytes, io_err := io.ReadAll(resp.Body)
+	assert.NilError(io_err)
+	msg := string(bodyBytes)
+	assert.Equal("id must be a positive number", msg)
+	assert.Equal(400, resp.StatusCode)
+
+	// create a GET request to the APP/api/v1/users/:id endpoint
+	req2 := httptest.NewRequest("GET", fmt.Sprintf("%s/api/v1/users/null", app.Address), nil)
+	resp2, err2 := app.App.Test(req2)
+	assert.NilError(err2)
+	defer resp2.Body.Close()
+	bodyBytes2, io_err2 := io.ReadAll(resp2.Body)
+	assert.NilError(io_err2)
+	msg2 := string(bodyBytes2)
+	assert.Equal("id must be a positive number", msg2)
+	assert.Equal(400, resp2.StatusCode)
+}
+func TestGetUserBadId(t *testing.T) {
```

Comment:
same idea as above, can we split into two tests, ex TestGetUserBadRequestNegativeID, TestGetUserBadRequestZeroID